src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1038,8 +1009,11 @@ unsigned SsaBuilder::RenamePushDef(GenTree* defNode, BasicBlock* block, unsigned
     assert(m_pCompiler->lvaInSsa(lclNum) && !m_pCompiler->lvaGetDesc(lclNum)->lvPromoted);
 
     LclVarDsc* const varDsc = m_pCompiler->lvaGetDesc(lclNum);
-    unsigned const   ssaNum =
-        varDsc->lvPerSsaData.AllocSsaNum(m_allocator, block, defNode->OperIs(GT_ASG) ? defNode->AsOp() : nullptr);
+    unsigned const   ssaNum = varDsc->lvPerSsaData.AllocSsaNum(m_allocator, block, defNode->OperIs(GT_ASG)
+#ifdef TARGET_WASM // for the second pass through SSA with LIR
+            || defNode->OperIs(GT_STORE_LCL_VAR)
+#endif // TARGET_WASM
+            ? defNode->AsOp() : nullptr);
```

Comment:
```suggestion
    unsigned const   ssaNum =
        varDsc->lvPerSsaData.AllocSsaNum(m_allocator, block, !defNode->IsCall() ? defNode->AsOp() : nullptr);
```
I should've written it this way originally...